Automattic/studio
# fix(sync): handle errors of deleted sites

## Related issues

Fixes https://github.com/Automattic/dotcom-forge/issues/9374

## Proposed Changes
With this PR we are handling case when a site was connected, then the user deleted it on wordpress.com, so we need to render error and Deleted state in Sync dialog

## Testing Instructions
1. Open Wordpress.com and purchase a site
2. Open Studio and connect it
3. Return to wordpress.com and delete the site purchased at the first step
4. Return to Studio and assert that you see error: <br /><img width="701" alt="Screenshot 2024-11-21 at 20 03 02" src="https://github.com/user-attachments/assets/6d050366-4271-458d-b2b7-3c8a7108cd5a">
5. Click on "Reconnect" button
6. Assert that you see that deleted site with "Deleted" state:<br ><img width="532" alt="Screenshot 2024-11-21 at 20 03 42" src="https://github.com/user-attachments/assets/c8645900-7b0f-4d9d-9331-8d12e2c9657b">
7. Select some another site in Sync dialog and click "Connect"
8. Assert that new site is connected; Deleted site disappeared; And if you open Sync dialog, you don't see that deleted site anymore
